ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReportingPage.tsx

```diff
@@ -1,16 +1,37 @@
 import React from 'react';
 import { Route, Switch } from 'react-router-dom';
 
-import { vulnerabilityReportingPath } from 'routePaths';
+import usePermissions from 'hooks/usePermissions';
+import { vulnerabilityReportingPath, vulnerabilityReportingCreatePath } from 'routePaths';
 
 import VulnReportsPage from './VulnReports/VulnReportsPage';
+import CreateVulnReportPage from './CreateVulnReport/CreateVulnReportPage';
 
 import './VulnReportingPage.css';
 
 function VulnReportingPage() {
+    const { hasReadWriteAccess, hasReadAccess } = usePermissions();
+
+    const hasWorkflowAdministrationWriteAccess = hasReadWriteAccess('WorkflowAdministration');
+    const hasImageReadAccess = hasReadAccess('Image');
+    const hasAccessScopeReadAccess = hasReadAccess('Access');
+    const hasNotifierIntegrationReadAccess = hasReadAccess('Integration');
+    const canReadWriteReports =
+        hasWorkflowAdministrationWriteAccess &&
+        hasImageReadAccess &&
+        hasAccessScopeReadAccess &&
+        hasNotifierIntegrationReadAccess;
```

Comment:
@md2119 @charmik-redhat 

Is there a plan to consolidate all the permissions necessary for report creation, that include all these needed perms, so we don't have such a compound conditional?

Reply:
Refer to [this document ](https://docs.google.com/document/d/1MeSyBdnDP9ldzci3lUoWCb6csHWHbYjD0_5uSglIOb4/) for details on the permission sets required for vulnerability management.

@sachaudh @vjwilson Can you explain what is `canReadWriteReports` used for?
Is it to hide the vulnerability report configuration list page?

Reply:
It'll be used to hide the buttle to create a report and also the page where you can create a report

Reply:
I am worried that we have to maintain permissions in two places. The permissions set evolved, and I can see UI using incorrect permission. I recommend not performing a permission check on the FE but let BE handle it.

Reply:
@md2119 
UI team has been asked by UX and Product to start tailoring the UI to current user's permissions. There is a 4.2 epic to do that in the current UI.
https://issues.redhat.com/browse/ROX-17061

Reply:
The idea is nice as long as the approach is solid. What is the plan to avoid discrepancies in the future?

Reply:
Two pronged approach:
1. We'll have ui-e2e tests to check our work, but most of the tests for limited permissions will have to be mocked for CI
2. From now on, any change to the permission system will need to be shared with the UI team, so that we can make any necessary changes. (no different than any other change to the API)

Reply:
(1) is okay, (2) is another manual process to maintain to the ever-increasing list of ACS processes which are prone to failure.

Do you want to check with the appropriate team to get a generic endpoint that returns auth pass/fail for all the ACS services? That way it is fully automated and all endpoints are handled.

Reply:
Not sure I understand this suggestion? How is it different than the `mypermisssions` endpoint we have now?

Reply:
`mypermisssions` endpoint does not tell you where or not you can access an API. It just informs about the permissions user has.